recipes/cpython/all/conanfile.py

```diff
@@ -236,11 +253,145 @@ def generate(self):
         else:
             self._generate_autotools()
 
-    def _patch_sources(self):
-        apply_conandata_patches(self)
+    def _msvc_project_path(self, name):
+        return os.path.join(self.source_folder, "PCBuild", f"{name}.vcxproj")
+
+    def _regex_replace_in_file(self, filename, pattern, replacement):
+        content = load(self, filename)
+        content = re.sub(pattern, replacement, content)
+        save(self, filename, content)
+
+    def _inject_conan_props_file(self, project_basename, dep_name, condition=True):
+        if condition:
+            search = '<Import Project="python.props" />'
+            replace_in_file(self,
+                            self._msvc_project_path(project_basename),
+                            search,
+                            search + f'<Import Project="{self.generators_folder}/conan_{dep_name}.props" />')
+
+    def _patch_setup_py(self):
         setup_py = os.path.join(self.source_folder, "setup.py")
         if Version(self.version) < "3.10":
             replace_in_file(self, setup_py, ":libmpdec.so.2", "mpdec")
+
+        if self.options.get_safe("with_curses", False):
+            # FIXME: this will link to ALL libraries of ncurses. Only need to link to ncurses(w) (+ eventually tinfo)
+            ncurses_info = self.dependencies["ncurses"].cpp_info.aggregated_components()
+            replace_in_file(self, setup_py,
+                "curses_libs = ",
+                "curses_libs = {} #".format(repr(ncurses_info.libs + ncurses_info.system_libs)))
+
+        if self._supports_modules:
+            openssl = self.dependencies["openssl"].cpp_info.aggregated_components()
+            zlib = self.dependencies["zlib"].cpp_info.aggregated_components()
+            if Version(self.version) < "3.11":
+                replace_in_file(self, setup_py,
+                                "openssl_includes = ",
+                                f"openssl_includes = {openssl.includedirs + zlib.includedirs} #")
+                replace_in_file(self, setup_py,
+                                "openssl_libdirs = ",
+                                f"openssl_libdirs = {openssl.libdirs + zlib.libdirs} #")
+                replace_in_file(self, setup_py,
+                                "openssl_libs = ",
+                                f"openssl_libs = {openssl.libs + zlib.libs} #")
+
+            if Version(self.version) < "3.11":
+                replace_in_file(self, setup_py, "if (MACOS and self.detect_tkinter_darwin())", "if (False)")
+
+    def _patch_msvc_projects(self):
```

Comment:
It's really hard to read all this, would be possible to covert to a patch instead?

Reply:
It would be hell indeed, they basically broke with every new patch version since they were so large. That was the motivation for switching to this method. There's an in-between option that I could explore which is to split the patches by file, in hopes that some of them could be re-used between versions. That might be more readable, but I don't know how much effort that would save us in the long run.

The option I _ideally_ wanted was fuzzy patching, but that seems to just not work, any insight would be appreciated there.

The injection of the dependency's `.prop`s files have to be done this way I think, since we can't configure the build type in the patch.

Reply:
It's not great but looks ok to me. Could use a few more comments for the reason for the patch for each block, if possible.

Reply:
Yea I think there's certainly a tradeoff here between readability and resilience to change. Can certainly revisit if fuzzy patching becomes an option again.

I can add some comments, oddly enough one of the issues with the original patches is that they're missing a lot of context as well.

Reply:
I added some comments, most things are "unvendor XYZ", there are a few things that I may need to take another look at.